tangelo/algorithms/variational/adapt_vqe_solver.py

```diff
@@ -93,6 +94,7 @@ def __init__(self, opt_dict):
         self.deflation_circuits: Optional[List[Circuit]] = copt_dict.pop("deflation_circuits", list())
         self.deflation_coeff: float = copt_dict.pop("deflation_coeff", 1)
         self.up_then_down: bool = copt_dict.pop("up_then_down", False)
+        self.spin: Union[int, None] = copt_dict.pop("spin", None)
```

Comment:
Should `None` be there ? Docs says `int`. Is there a better default value than `None` ?
Could it be `0` instead ?

Reply:
`n_electron` and `n_spinorbitals` are still None to conform with `fermion_to_qubit_mapping` arguments and checks.

Reply:
That is ok and whatever we decide to do about this in the future is up to us.
Ideally they should be integers but this is more of a "codebase fix" than something actually needed atm.